### PR TITLE
Create unified AMI creation role

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR creates a single, custom role that contains all permissions needed to create an AMI (including read access to specified parameter store parameters) and can be assumed by the user created by this module.

## 💭 Motivation and Context
Previously, this module created a user with the ability to assume two different roles:
 * One standard role (`EC2AMICreate`) with permissions to create the AMI 
 * Another custom role (`ParameterStoreReadRole-*`) with permissions to read parameter store parameters

That proved to be a problem, since we create AMIs with `packer`, which expects all permissions to be included in the role that is assumed when `packer` is launched.  By consolidating all of the permissions into a single, custom role for this user, `packer` is able to successfully create AMIs.

Note that the `ParameterStoreReadRole-*` is still created (no changes to it in this PR), since it will be needed in use cases where AMI-creation permissions are not required.

See related PR https://github.com/cisagov/cool-accounts/pull/21.

## 🧪 Testing
* The example test user was created and reviewed in the AWS console for accuracy.
* I created the `test-openvpn-packer` user (using updated code in https://github.com/cisagov/openvpn-packer; PR coming soon with those changes) and verified that I was able to successfully create an AMI with that user.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by jsf9k — CHANGES_REQUESTED** [review]
For molecule, we want to create roles that simply read one or more SSM Parameter Store parameters. 
 With the current code, you will have to provide a provider that doesn't get used if you specify `var.add_packer_permissions = false`.  This is ugly and troubles me greatly.

I'm thinking that maybe this module is doing too much, and maybe we need to decompose it into:
* A module that creates a policy for reading specified SSM Parameter Store parameters
* A module that leverages the previous module and goes on to create a user that can assume the role.  This module is useful for testing molecule repositories via CI.
* This module, which leverages the first module but also creates the AMI creation policy, a role with both policies attached, and a user that can assume that role.

I don't want to ask you to do the decomposition right now, since you have just run a gauntlet getting cross-account AMI creation to work, but we should at least create an issue so I can circle back to this later.  If everyone agrees with the approach, and the issue is created, then I will approve this pull request.

**Comment by jsf9k** [comment]
@dav3r agreed with the idea of decomposing this module, so I created issue #10.

**Review by jsf9k — APPROVED** [review]

File: locals.tf [hunk 1]
```diff
@@ -2,6 +2,16 @@
 data "aws_caller_identity" "current" {}
 
 locals {
+  # If var.ec2amicreate_role_description contains "%s", use format()
+  # to replace the "%s" with var.user_name, otherwise just use
+  # var.ec2amicreate_role_description as is
+  ec2amicreate_role_description = length(regexall(".*%s.*", var.ec2amicreate_role_description)) > 0 ? format(var.ec2amicreate_role_description, var.user_name) : var.parameterstorereadonly_role_description
```
Comment (felddy) [inline locals.tf:8]: Cool. 👍

**Review by felddy — APPROVED** [review]
Nice work!  🐕 🚶 🤖

**Pull request merged**